### PR TITLE
Adds a POST /v1/zonefile API endpoint

### PR DESCRIPTION
**Pull request by kantai**
This PR implements two features:

1. Adds an API endpoint `POST /v1/zonefile` -- which allows a client to attempt to publish a zonefile whose hash has already been included in a confirmed blockchain transaction.
2. Adds an ENV var to control an integration test setting for bitcoind -- this is required for connecting to the bitcoind node if its running inside of a container.

### CODE REVIEW

File: blockstack_client/rpc.py [hunk 1]
```diff
@@ -1040,6 +1040,43 @@ def PUT_name_transfer( self, ses, path_info, name ):
         self._reply_json(resp, status_code=202)
         return
 
+    def POST_raw_zonefile( self, ses, path_info ):
+        """
+        Publish a zonefile which has *already* been announced.
+        Return 200 and {'status': True, 'servers': ...} on success
+        Return 401 when zonefile invalid with {'error': ...}
+        """
+        request_schema = {
```
Comment (jcnelson) [inline blockstack_client/rpc.py]: Zone files can be arbitrary data in practice.  Blockstack Core requires any uploaded zone file data to be base64-encoded, in order to accommodate this.  Should this endpoint also accept arbitrary zone file data?  Or is this endpoint only intended for "well-formed" zone files that are guaranteed to be ASCII (or at a minimum, guaranteed to be JSON-serializable)?

File: blockstack_client/rpc.py [hunk 2]
```diff
@@ -3300,6 +3337,20 @@ def _dispatch(self, method_name):
                     },
                 },
             },
+            r'^/v1/zonefile$': {
+                'routes': {
+                    'POST' : self.POST_raw_zonefile
+                },
+                'whitelist': {
+                    'POST': {
```
Comment (jcnelson) [inline blockstack_client/rpc.py]: These security settings will mean that anyone in the world can use this endpoint to broadcast a zone file.  Just checking that this the intended behavior?  I can see it for e.g. core.blockstack.org, for example.

Comment (kantai) [reply blockstack_client/rpc.py]: Yep, the intention is for this to be a public endpoint.

File: blockstack_client/rpc.py (hunk 1 above)
Comment (kantai) [reply blockstack_client/rpc.py]: Sure, this can support b64

**Comment by kantai** [comment]
Thanks for reviewing this @jcnelson -- I'm adding support for arbitrary b64 encoded data and an integration test now.

**Comment by kantai** [comment]
Oh -- I missed this comment before:

> have blockstack.js do the work in zonefile_data_publish? This would obviate the need for this PR.

I think the conclusion of 

https://github.com/blockstack/blockstack/blob/future-architecture/future-architecture.md

Was that blockstack.js would communicate with blockstackd via the API endpoints, rather than using the RPC interface directly.

**Comment by jcnelson** [comment]
Ah okay.  In that case, I'll start making a list of methods from `blockstack_client` that will need to be preserved in order to implement the stable API endpoint service.

**Comment by jcnelson** [comment]
Tested and confirmed it works.  Looks good to me.  Merging...

**Pull request merged**